test/k8s/service_helpers.go

```diff
@@ -218,12 +219,12 @@ func testCurlFromOutsideWithLocalPort(kubectl *helpers.Kubectl, ni *helpers.Node
 			"Can not connect to service %q from outside cluster (%d/%d)", url, i, count)
 		if checkSourceIP {
 			// Parse the IPs to avoid issues with 4-in-6 formats
-			sourceIP := net.ParseIP(strings.TrimSpace(strings.Split(res.Stdout(), "=")[1]))
-			var outIP net.IP
-			if sourceIP.To4() != nil {
-				outIP = net.ParseIP(ni.OutsideIP)
+			sourceIP := netip.MustParseAddr(strings.TrimSpace(strings.Split(res.Stdout(), "=")[1]))
```

Comment:
Please use ParseAddr and check return error value with `ExpectWithOffset`. `MustParseAddr` will panic, which is not ideal.